### PR TITLE
Suggestfunction

### PR DESCRIPTION
**Pull request by jcriotte**
Delivery suggestion function after merge with dev

### CODE REVIEW

File: src/Controller/PostController.php [hunk 1]
```diff
@@ -65,8 +68,41 @@ public function search(): string
...
+            $key = false;
+            foreach ($searchs as $searchtab) {
+                if (array_search($searchInput['maRecherche'], $searchtab)) {
+                    $key = true;
+                    // si oui alors mise à jour des données date et nb_searched
+                    $updateSearch = [];
+                    $updateSearch['id'] = $searchtab['id'];
+                    $updateSearch['date_last'] = date("Y-m-d H:i:s");
+                    $updateSearch['nb_searched'] = $searchtab['nb_searched'] + 1;
+                    $search = new SearchManager();
```
Comment (aldaitzwild) [inline src/Controller/PostController.php:84]: Why do you use a second searchManager ??? you already have one at this point (look line 72)

File: src/Controller/PostController.php [hunk 2]
```diff
@@ -65,8 +68,41 @@ public function search(): string
...
+                    break;
+                }
+            }
+            if (!$key) {
+                // si non alors création d'un nouvel enregistrement dans la table search
+                $newSearch = [];
+                $newSearch['word'] = $searchInput['maRecherche'];
+                $newSearch['date_last'] = date("Y-m-d H:i:s");
+                $newSearch['nb_searched'] = 1;
+                $search = new SearchManager();
```
Comment (aldaitzwild) [inline src/Controller/PostController.php:98]: And a third one ???

**Review by aldaitzwild — CHANGES_REQUESTED** [review]

**Comment by jcriotte** [comment]
PostController Fix Delivery and Layout Change Following Regression

File: src/Controller/PostController.php [hunk 3]
```diff
@@ -65,8 +68,40 @@ public function search(): string
             // if validation is ok, insert and redirection
             $postManager = new PostManager();
             $results = $postManager->search($searchInput['maRecherche']);
+            // traitement of search
+            $searchManager = new SearchManager();
+            $searchs = $searchManager->selectAll();
+            // rechercher si le mot cle est dans la table search
+            $key = false;
+            $search = new SearchManager();
```
Comment (aldaitzwild) [inline src/Controller/PostController.php:76]: Still the same problem ... Why do you instanciate a second SearchManager() ? There is no mean for that

**Review by aldaitzwild — CHANGES_REQUESTED** [review]

File: src/Controller/PostController.php [hunk 4]
```diff
@@ -3,6 +3,9 @@
 namespace App\Controller;
 
 use App\Model\PostManager;
+use App\Model\SearchManager;
+
+use function Amp\Internal\getCurrentTime;
```
Comment (aldaitzwild) [inline src/Controller/PostController.php:8]: What is this use ?? You doesn't seems to use it

**Review by aldaitzwild — CHANGES_REQUESTED** [review]

**Comment by jcriotte** [comment]
Delivery PostController with out this line.

**Review by aldaitzwild — APPROVED** [review]

**Pull request merged**